### ISSUE TITLE
 Cache-Control HTTP Headers for File-Generating Endpoints

### ISSUE THREAD

**Issue by Sibyx** [EvilFlowersCatalog/EvilFlowersCatalog#29]
It would be beneficial to have the ability to set Cache-Control HTTP headers for endpoints that generate files (e.g., apps.files endpoints). Please consider whether this functionality should be implemented on the reverse proxy (nginx) or within the Django application itself.

While implementing this on nginx seems logical, having all cache settings centralized within a single layer could simplify usability and configuration. Currently, we use nginx for server-side caching of these files.

To proceed, please decide on unified cache policies or propose a framework for organizing cache layers. Document these decisions comprehensively in a detailed wiki or similar resource.

**Comment by Sibyx** [comment]
## Decision

The application is the source of truth for `Cache-Control` on file-serving endpoints. nginx (and any CDN) must respect the upstream headers and not rewrite them. This keeps the policy in one place (Django settings, version-controlled) instead of being split between two repos with the proxy silently overriding the app.

## App-layer implementation (this repo)

Two settings drive the policy, both tunable per-environment via env vars:

| Setting | Default | Used for |
|---|---|---|
| `EVILFLOWERS_FILES_CACHE_CONTROL_PUBLIC` | `public, max-age=86400` | `EntryImageDownload`, `EntryThumbnailDownload` |
| `EVILFLOWERS_FILES_CACHE_CONTROL_PRIVATE` | `private, max-age=0, must-revalidate` | `AcquisitionDownload`, `UserAcquisitionDownload` |

Why split: cover images and thumbnails are large, immutable for the lifetime of the entry, and not user-specific — so they're great cache targets at any layer. Acquisitions can carry user-specific watermarks (`apps/core/modifiers/`) and personalisation, so they must not be cached on shared caches.

Readium encrypted-content downloads already emit `private, no-store` (`apps/readium/views/content.py:71`); that endpoint stays as-is since the no-store guarantee is part of the LCP compliance posture (see `apps/readium/tests/test_lcp_compliance.py`).

## nginx / infrastructure follow-up

Tracked in EvilFlowersCatalog/elvira-devops#1 — covers:

- nginx config respects upstream `Cache-Control` (no rewriting / stripping)
- nginx-level `proxy_cache` for `public` cover/thumbnail responses (recommended)
- Explicit `proxy_no_cache` / `proxy_cache_bypass` for `private` acquisition responses
- CDN behaviour, if applicable

Closing this issue with the app-layer change in place. Once the devops follow-up lands, the policy is fully reconciled across both layers.

**Comment by Sibyx** [comment]
Correction: the devops follow-up issue is EvilFlowersCatalog/elvira-devops#5 (not #1 as referenced above).
